src/Core/IThreadHandling.cs

```diff
@@ -62,6 +63,24 @@ public interface IThreadHandling
         /// Switches to the background thread
         /// </summary>
         /// <remarks>Wrapper that calls <see cref="TaskScheduler.Default"/></remarks>
-        Task SwitchToBackgroundThread();
+        IAwaitableWrapper SwitchToBackgroundThread();
+    }
+
+    // Wrappers for awaiter /awaitable to avoid VS-specific types on the interface
+    public interface IAwaitableWrapper
+    {
+        IAwaiterWrapper GetAwaiter();
+    }
+
+    public interface IAwaiterWrapper :
+#if VS2022
+        // Earlier versions of VS don't implement ICriticalNotifyCompletion
```

Comment:
I didn't realise this until the PR build failed for VS2017 and VS2019.
I looked at the VS2019 version in DotPeek to check that it only implements the non-critical version.